sky/serve/serve_utils.py

```diff
@@ -879,9 +886,10 @@ def _build(cls, code: List[str]) -> str:
         return f'python3 -u -c {shlex.quote(generated_code)}'
 
     @classmethod
-    def update_service(cls, service_name: str, version: int) -> str:
+    def update_service(cls, service_name: str, version: int, mode: str) -> str:
```

Comment:
Is there any back-compat issue? E.g., old serve controller, client update past this PR?

Reply:
Good point! Added the handling for backward compatibility. : )